docs/using/administering/debugging.rst

```diff
@@ -119,6 +119,47 @@ Alternatively, the information can be gathered manually using the following comm
 
 * Flocker log files (see :ref:`Flocker logging <flocker-logging>` above)
 
+Profiling
+---------
+
+.. warning::
+
+   It is not recommended to use profiling while relying on Flocker within a production environment as there is a performance overhead.
+
+Control Service
+^^^^^^^^^^^^^^^
+
+It is possible to obtain :py:mod:`cProfile` profiling data of the :ref:`Control Service <control-service>` between two user defined intervals.
+
+Profiling is disabled by default.
+To enable profiling of the Control Service run the following command as root on the control node:
+
+.. prompt:: bash #
+
+   pkill -SIGUSR1 flocker-control
+
+Profiling data will then be collected until a signal to dump the profiling data is received.
+To dump the current profile run the following command as root on the control node:
+
+.. prompt:: bash #
+
+   pkill -SIGUSR2 flocker-control
+
+This will output the profiling data to a file named :file:`/etc/flocker/profile-<TIMESTAMP>`.
+This will also disable profiling.
```

Comment:
:question: Change this to emphasise first that SIGUSR1 starts profiling and SIGUSR2 stops profiling. Then mention that when profiling is stopped, all profiling data collected up to that point, including from previous intervals of profiling, will be dumped. This clearly matches the two signals as delimiting phases of profiling, while also explaining how to get the profiling data.